libultraship/libultraship/ImGuiImpl.cpp

```diff
@@ -338,7 +346,36 @@ namespace SohImGui {
         ImGui::SetCurrentContext(ctx);
         io = &ImGui::GetIO();
         io->ConfigFlags |= ImGuiConfigFlags_DockingEnable;
-        io->Fonts->AddFontDefault();
+        #ifdef _WIN32
+		    ::SetProcessDPIAware();
+            ImFontConfig imConfig;
+            UINT dpi = GetDpiForSystem();
+            int points = 12;
+            float size = ceilf(points * dpi / 72.f);
+            dpi_scale = ceilf(dpi / 72.f);
+            imConfig.SizePixels = size;
+            io->Fonts->AddFontDefault(&imConfig);
+        #elif __linux_
+                std::array<char, 128> buffer;
+                std::string result;
+                std::unique_ptr<FILE, decltype(&pclose)> pipe(popen("xrdb -query |grep dpi", "r"), pclose);
```

Comment:
yikes, popen? can you use the X (or wayland if detected) apis instead?

Reply:
sure, just testing but can look into X apis, wayland may be difficult